cli/cmd/proxy/list/command.go

```diff
@@ -150,23 +148,27 @@ func (c *ListCommand) initKubernetes() error {
 		return fmt.Errorf("error creating Kubernetes client %v", err)
 	}
 
+	return err
+}
+
+func (c *ListCommand) namespace() string {
+	settings := helmCLI.New()
+
 	if c.flagAllNamespaces {
-		c.namespace = "" // An empty namespace means all namespaces.
+		return "" // An empty namespace means all namespaces.
 	} else if c.flagNamespace != "" {
-		c.namespace = c.flagNamespace
+		return c.flagNamespace
 	} else {
-		c.namespace = settings.Namespace()
+		return settings.Namespace()
 	}
-
-	return err
 }
```

Comment:
I pulled `namespace` out into its own method because when it was part of `initKubernetes`, which only runs when `c.kubernetes` is `nil`, the `namespace` being passed in was ignored during tests as they patch in a fake Kubernetes client to `c.kubernetes`, bypassing the method call.